Steps towards GOG integration

NicoVIII opened the pull request:
As mentioned in #154 I really wanted to help implementing proper GOG integration. Because I didn't get an answer to my suggestions yet, I could not wait and implemented some stuff which worked fine for me.

I mostly changed the interpreter to support gog games with those easily. The structure I used should be changeable to whatever the maintainers want, I implemented just a first suggestion.
If lutris is not connected to gog yet, it will ask for connection during the installation process. After that it will download the defined installer (it's imaginable to use other parameter like os and language to choose an installer instead of the id) and extract it in the way described in the wiki to install gog linux games, if the os of the installer is "linux". Otherwise the installation should be controlled in the usual way.

I used two installer scripts to test the behaviour. Those should be easily adjustable for other gog games.

GOG native linux game:
```
id: 9499
game: "http://lutris.net/api/games/this-is-the-police?format=json"
game_slug: "this-is-the-police"
name: "This Is the Police"
year: null
user: "NicoVIII"
runner: "linux"
slug: "this-is-the-police-gog"
version: "GOG"
description: "Test!"
notes: ""
created_at: "2018-07-31T15:22::57.296000Z"
updated_at: "2018-08-01T05:00:44.690704Z"
draft: true
published: true
rating: ""
steamid: 443810
gogslug: ""
humblestoreid: ""
script:
  game:
    gog:
      appid: 1513553405
      installerid: installer_linux_en
```

GOG wine game:
```
id: 6169
game: "http://lutris.net/api/games/lula-the-sexy-empire?format=json"
game_slug: "lula-the-sexy-empire"
name: "Lula: The Sexy Empire"
year: 1998
user: "FakeShemp"
runner: "wine"
slug: "lula-the-sexy-empire-gog"
version: "GOG"
description: ""
notes: ""
created_at: "2017-10-29T07:07:19.122000Z"
updated_at: "2017-10-29T07:27:06.859898Z"
draft: false
published: true
rating: ""
steamid: null
gogslug: "lula_the_sexy_empire"
humblestoreid: ""
script:
  game:
    gog:
      appid: 1443433575
      installerid: installer_windows_en
    exe: "drive_c/GOG Games/Lula - The Sexy Empire/WET.EXE"
    prefix: "$GAMEDIR"
  installer:
    - task:
        args: "/sp- /verysilent /norestart /suppressmsgboxes /nogui"
        name: "wineexec"
        executable: "$installer"
        prefix: "$GAMEDIR"
    - task:
        name: "winetricks"
        app: "icodecs"
        prefix: "$GAMEDIR"
    - task:
        name: "winetricks"
        app: "quartz"
        prefix: "$GAMEDIR"
```

I hope those changes can be a solid base for more steps towards GOG integration.

strycore commented:
Awesome! There seems to be some nice stuff in here but could you rebase into a single (or a few meaningful commits) it so it's clearer what the patchset actually does and easier to review? Thanks!

NicoVIII commented:
Of course, no problem!

NicoVIII commented:
I rebased it into a single commit. In this commit I saw, that I changed (out of a habbit) some small things which have nothing to do with gog integration in general (fix typo in comment, rename not used variable to _ etc.).
Do you think something like that is important enough to create a single pull request, is it okay to provide it in this one or should I remove that completely?

strycore commented:
@NicoVIII thanks! You can leave the minor changes in this PR, no problem.

I was going to suggest to remove the type hints since we also target python 3.4 (and maybe below)

NicoVIII commented:
Alright, I hope I found all, but I removed the two typehints I found.
